src/ui/public/notify/notifier.js

```diff
@@ -97,10 +97,14 @@ function add(notif, cb) {
     });
   } else if (notif.customActions) {
     // wrap all of the custom functions in a close
-    notif.customActions = notif.customActions.map(action => {
+    notif.customActions = notif.customActions.map((action, idx) => {
       return {
         key: action.text,
-        callback: closeNotif(notif, action.callback, action.text)
+        callback: closeNotif(notif, action.callback, action.text),
+        getButtonClass() {
+          if (idx === 0) return `btn-primary btn-${notif.type}`;
+          return `btn-default btn-${notif.type}`;
+        }
```

Comment:
I think we can make this a little more readable, maintainable (by making the class names easier to grep for, and also uncoupling them from the `notif.type`), and clearer. What do you think of this? 

```javascript
// Not sure if this is the best name for it, since I'm not sure what the role/meaning of `idx` is.
let buttonBaseClass;

if (idx === 0) {
  buttonBaseClass = 'btn-primary';
} else {
  buttonBaseClass = 'btn-default';
}

// This type of map explicitly documents the possible classes close to the logic which will generate them.
// I think this improves readability, and makes it easier to grep for class names like "btn-danger".
// More importantly, this also uncouples the notification types from the class names we use.
const typeToButtonClassMap = {
  error: 'btn-danger',
  warning: 'btn-warning',
  info: 'btn-info',
};

const buttonTypeClass = typeToButtonClassMap[notif.type];

return `${buttonBaseClass} ${buttonTypeClass}`;
```